app-grpc-test/src/test/java/net/spals/appbuilder/app/grpc/GrpcWebAppWithMocksFTest.java

```diff
@@ -0,0 +1,52 @@
+package net.spals.appbuilder.app.grpc;
+
+import net.spals.appbuilder.app.examples.grpc.sample.SampleGrpcWebApp;
+import net.spals.appbuilder.filestore.core.FileStore;
+import net.spals.appbuilder.mapstore.core.MapStore;
+import org.mockito.internal.util.MockUtil;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Functional tests for a sample {@link GrpcWebAppWithMocks}
+ *
+ * @author tkral
+ */
+public class GrpcWebAppWithMocksFTest {
+
+    private final GrpcWebApp sampleAppWithMocks = GrpcWebAppWithMocks.grpcWithMocks(new SampleGrpcWebApp())
+        .addMockSingleton(mock(MapStore.class), MapStore.class);
+    private final GrpcTestSupport testServerWrapper = GrpcTestSupport.embeddedGrpc(sampleAppWithMocks);
+
+    @BeforeClass
+    void classSetup() {
+        testServerWrapper.before();
+    }
+
+    @AfterClass
+    void classTearDown() {
+        testServerWrapper.after();
+        assertThat(sampleAppWithMocks.isRunning(), is(false));
+    }
+
+    @Test
+    public void testInjectedMocks() {
+        // Assert that the mocked MapStore is present in the service graph.
+        assertThat(MockUtil.isMock(
```

Comment:
Out of scope, but I'm wondering if you use isMock and would you want a custom matcher.

i.e. `assertThat(foo, isMock())` would read more fluently 

Are you in the habit of making your own matchers? I can show you how if you are unfamiliar.

Reply:
I'm familiar, but not in the habit of doing it. In this case, I skip just because I didn't think I'd use the matcher very often.